common/include/pcl/for_each_type.h

```diff
@@ -90,7 +90,8 @@ namespace pcl
   template<typename Sequence, typename F> inline void 
   for_each_type (F f)
   {
-    BOOST_MPL_ASSERT (( boost::mpl::is_sequence<Sequence> ));
```

Comment:
Can we remove the include https://github.com/PointCloudLibrary/pcl/pull/5525/files#diff-8032fb4db6d8df89f677ff5ad3196a07f99aa862d60784a6a2c89f079a63cfb6L50 a well in this file?

Reply:
Do you mean the Eigen includes? I don't know, because it has nothing to do with the Boost macro directly. I only see that the reference to `line 84` is no longer correct, but would be line 80 by now (so it was already wrong before the PR).

Reply:
Hmm.
Not sure why the link goew to the wrong line. It is the inclusionnof boost_mpl_assert in for_each_type.

Reply:
Removed the include now (and fixed the one comment as well - thank for the wrong link 😆). There are some more includes we could remove (`is_sequence.hpp` e.g. already includes `begin_end.hpp` and `not.hpp`), but as I can't test all supported versions of Boost, I don't touch this in this MR (we should use modules of C++20^^).